rclcpp/include/rclcpp/executor.hpp

```diff
@@ -244,9 +245,16 @@ class Executor
     }
     std::chrono::nanoseconds timeout_left = timeout_ns;
 
-    while (rclcpp::ok(this->context_)) {
+    if (spinning.exchange(true)) {
+      throw std::runtime_error("spin_some() called while already spinning");
+    }
+    RCLCPP_SCOPE_EXIT(this->spinning.store(false); );
+    while (rclcpp::ok(this->context_) && spinning.load()) {
       // Do one item of work.
-      spin_once(timeout_left);
+      AnyExecutable any_exec;
+      if (get_next_executable(any_exec, timeout_left)) {
+        execute_any_executable(any_exec);
+      }
```

Comment:
Why change this?

Reply:
If use spin_once, when finishing spin_once, spinning value will change as false.